File: code/backend/webflux/v5/customer/webflux_customerprice_service/src/main/java/at/bumzack/foryouandyourfakewebshop/searchindexindex/model/CustomerPriceEntry.java
package at.bumzack.foryouandyourfakewebshop.searchindexindex.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;


@Table("customer_price")
public record CustomerPriceEntry(@Id Long id, Long customerId, Double discount, Integer startYear, Integer endYear, LocalDateTime created) {
}

File: code/backend/csharp/CustomerPriceService/CustomerPriceService/CustomerPriceEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CustomerService;


[Table("customer_price")]
public class CustomerPriceEntry
{

    [Key, Required]
    public long Id { get; set; }

    public long CustomerId  { get; set; }

    public double? Discount { get; set; }
    
    public int? StartYear { get; set; }

    public int? EndYear { get; set; }

    public DateTime? Created { get; set; }
}
